src/rgw/rgw_trim_bilog.cc

```diff
@@ -548,9 +705,15 @@ int BucketTrimInstanceCR::operate(const DoutPrefixProvider *dpp)
     min_markers.assign(std::max(1u, pbucket_info->layout.current_index.layout.normal.num_shards),
                        RGWSyncLogTrimCR::max_marker);
 
+    // Determine the minimum generation
+    retcode = take_min_generation();
+    if (retcode < 0) {
+      ldpp_dout(dpp, 4) << "failed to find minimum generation" << dendl;
+      return set_cr_error(retcode);
+    }
     // determine the minimum marker for each shard
-    retcode = take_min_status(cct, peer_status.begin(), peer_status.end(),
-                              &min_markers);
+    retcode = take_min_status(cct, totrim.gen, peer_status.cbegin(),
+			      peer_status.cend(), &min_markers);
```

Comment:
note that `min_markers` is initialized above based on `pbucket_info->layout.current_index.layout.normal.num_shards`, but we'll instead need to use the `num_shards` from the log generation we're trimming

Reply:
Fixed.